### PR TITLE
Rename vmset, overcloudnet, overcloudipset, provisionserver

### PR DESCRIPTION
**Pull request by dprince**

### CODE REVIEW

File: api/v1beta1/openstackbaremetalset_types.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ type OpenStackBaremetalSetSpec struct {
 	Replicas int `json:"replicas,omitempty"`
 	// Remote URL pointing to desired RHEL qcow2 image
 	RhelImageURL string `json:"rhelImageUrl,omitempty"`
-	// ProvisionServerName Optional. If supplied will be used as the base Image for the baremetalset instead of RhelImageURL.
+	// OpenStackProvisionServerName Optional. If supplied will be used as the base Image for the baremetalset instead of RhelImageURL.
 	ProvisionServerName string `json:"provisionServerName,omitempty"`
```
Comment (abays) [inline api/v1beta1/openstackbaremetalset_types.go:34]: Looks like this needs to be changed from `ProvisionServerName` to `OpenStackProvisionServerName`, given the comment above

File: config/crd/bases/osp-director.openstack.org_openstackbaremetalsets.yaml [hunk 2]
```diff
@@ -147,8 +147,8 @@ spec:
                 secret data'
               type: string
             provisionServerName:
```
Comment (abays) [inline config/crd/bases/osp-director.openstack.org_openstackbaremetalsets.yaml:149]: Should the description say `ProvisionServerName` or `OpenStackProvisionServerName`?  Not sure whether the key name or the comment name is right.

File: config/crd/bases/osp-director.openstack.org_openstackprovisionservers.yaml [hunk 3]
```diff
@@ -6,20 +6,21 @@ metadata:
...
+    listKind: OpenStackProvisionServerList
+    plural: openstackprovisionservers
+    singular: openstackprovisionserver
   scope: Namespaced
   subresources:
     status: {}
   validation:
     openAPIV3Schema:
-      description: ProvisionServer is the Schema for the provisionservers API
+      description: OpenStackProvisionServer is the Schema for the provisionservers
```
Comment (abays) [inline config/crd/bases/osp-director.openstack.org_openstackprovisionservers.yaml:22]: Should `provisionserver` here and below be renamed?

File: config/crd/bases/osp-director.openstack.org_openstackvmsets.yaml [hunk 4]
```diff
@@ -35,7 +35,7 @@ spec:
         metadata:
           type: object
         spec:
-          description: VMSetSpec defines the desired state of VMSet
+          description: OpenStackVMSetSpec defines the desired state of VMSet
```
Comment (abays) [inline config/crd/bases/osp-director.openstack.org_openstackvmsets.yaml:38]: Should `VMSet` here and below be renamed?

File: config/manifests/bases/osp-director-operator.clusterserviceversion.yaml [hunk 5]
```diff
@@ -6,48 +6,58 @@ metadata:
...
+      name: openstackvmsets.osp-director.openstack.org
+      version: v1beta1
+    - description: OpenStackProvisionServer is the Schema for the provisionservers API
+      displayName: OpenStack Provision Server
+      kind: ProvisionServer
+      name: openstackprovisionservers.osp-director.openstack.org
+      version: v1beta1
     - description: OpenStackBaremetalSet represent a set of baremetal hosts for a specific role within the Overcloud deployment
-      displayName: OpenStackBaremetal Set
+      displayName: Open Stack Baremetal Set
```
Comment (abays) [inline config/manifests/bases/osp-director-operator.clusterserviceversion.yaml:28]: Is the spacing intended here and for `Open Stack Control Plane` below?

**Review by stuggi — COMMENTED** [review]
Do we want to add shortNames as described in [1] via this rename, or in a follow up?

[1] https://github.com/openstack-k8s-operators/osp-director-operator/issues/141

File: config/rbac/role.yaml [hunk 6]
```diff
@@ -261,7 +261,7 @@ rules:
 - apiGroups:
   - osp-director.openstack.org
   resources:
-  - overcloudipsets
+  - openstackipset
```
Comment (abays) [inline config/rbac/role.yaml:264]: Does this and its other references in this file need to be plural?  I just see us using the plural elsewhere.

File: controllers/openstackvmset_controller.go [hunk 7]
```diff
@@ -86,17 +86,17 @@ func (r *VMSetReconciler) GetScheme() *runtime.Scheme {
 // FIXME: Is there a way to scope the following RBAC annotation to just the "openshift-machine-api" namespace?
 // +kubebuilder:rbac:groups=kubevirt.io,resources=virtualmachines,verbs=list;watch
 // +kubebuilder:rbac:groups=nmstate.io,resources=nodenetworkconfigurationpolicies,verbs=create;delete;get;list;patch;update;watch
-// +kubebuilder:rbac:groups=osp-director.openstack.org,resources=overcloudipsets,verbs=get;list;watch;create;update;patch;delete
-// +kubebuilder:rbac:groups=osp-director.openstack.org,resources=overcloudipsets/finalizers,verbs=get;list;watch;create;update;patch;delete
-// +kubebuilder:rbac:groups=osp-director.openstack.org,resources=overcloudipsets/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=osp-director.openstack.org,resources=openstackipset,verbs=get;list;watch;create;update;patch;delete
```
Comment (abays) [inline controllers/openstackvmset_controller.go:89]: Might need plural here, but I'm not sure

File: config/crd/bases/osp-director.openstack.org_openstackbaremetalsets.yaml (hunk 2 above)
Comment (dprince) [reply config/crd/bases/osp-director.openstack.org_openstackbaremetalsets.yaml:149]: I would going to leave this just 'provisionServerName' for brevity.

Comment (abays) [reply config/crd/bases/osp-director.openstack.org_openstackbaremetalsets.yaml:149]: Ack

File: controllers/openstackvmset_controller.go (hunk 7 above)
Comment (dprince) [reply controllers/openstackvmset_controller.go:89]: ack. will change.

**Comment by dprince** [comment]
> Do we want to add the shortNames tags as described in [1] via this rename, or in a follow up?
> 
> [1] #141

I was thinking of doing this as a follow up.

File: config/rbac/role.yaml (hunk 6 above)
Comment (dprince) [reply config/rbac/role.yaml:264]: got it in the next rebase.

File: config/manifests/bases/osp-director-operator.clusterserviceversion.yaml (hunk 5 above)
Comment (dprince) [reply config/manifests/bases/osp-director-operator.clusterserviceversion.yaml:28]: This is auto generated. Let me see if I can fix it.

File: config/crd/bases/osp-director.openstack.org_openstackprovisionservers.yaml (hunk 3 above)
Comment (dprince) [reply config/crd/bases/osp-director.openstack.org_openstackprovisionservers.yaml:22]: I wasn't going to change these parameter names as part of this patch.

Comment (dprince) [reply config/crd/bases/osp-director.openstack.org_openstackprovisionservers.yaml:22]: Perhaps we can debate the pros/cons of this?

Comment (dprince) [reply config/crd/bases/osp-director.openstack.org_openstackprovisionservers.yaml:22]: oh, I see. This is just a comment. So yeah, lets fix it!

File: config/crd/bases/osp-director.openstack.org_openstackvmsets.yaml (hunk 4 above)
Comment (dprince) [reply config/crd/bases/osp-director.openstack.org_openstackvmsets.yaml:38]: Yes. I will fix this (in the types.go file..) and regenerate the CRD

File: api/v1beta1/openstackbaremetalset_types.go (hunk 1 above)
Comment (dprince) [reply api/v1beta1/openstackbaremetalset_types.go]: I updated the comment so this stays the same (ProvisionServerName).

**Review by stuggi — APPROVED** [review]
/lgtm

**Pull request merged**